SHARD-913 Added ENV for config and Docker image building workflow #74

arhamj opened the pull request.

arhamj committed:
ci(docker): added docker workflow and updated some config variables to env

fix(workflow): docker push workflow fix

feat(ci): updateing to use tags got actions instead of commit hashes

fix(docker): updated the actions to use pat

feat(actions): print env variables

aniketdivekar commented on .github/workflows/docker.yml:
```diff
@@ -0,0 +1,54 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Display IMAGE_NAME
+        run: echo "IMAGE_NAME is ${{ env.IMAGE_NAME }}"
```
Do we have tag w/ image name? or we are just overwriting one image?

arhamj replied:
Tag is defined in the section`Build and push Docker image`. The convention after discussion with devops is to use {branch-short_commit_hash}

arhamj committed:
ci(docker): updated docker workflow to simplify

chrypnotoad commented:
@arhamj @aniketdivekar This patch no longer applies because of the config.ts changes
`cache-test-settings.patch`

arhamj committed:
fix(patches): updated local setting patches

arhamj committed:
ci(workflow): addded a dispatcher trigger for docker image build

arhamj commented:
> @arhamj @aniketdivekar This patch no longer applies because of the config.ts changes `cache-test-settings.patch`

@chrypnotoad I have fixed the patches for now. Ideally, I would have wanted to delete these patches in favour of using env variables but, I don't have a solution to env management for local just yet. Thanks for catching it early. I've also added a `workflow_dispatcher` trigger as per the suggestion but since this is a public repo I'm not sure if anyone will be able to trigger this or if it's only members who can do it etc.

chrypnotoad commented on .github/workflows/docker.yml:
```diff
@@ -0,0 +1,67 @@
+      tag:
+        description: 'Tag for the Docker image'
+        required: false
+        default: 'latest' # Default tag if none is provided
```
this input isn't being used when the tag is set

arhamj replied:
Updated.

arhamj committed:
ci(docker): use custom tag for manual workflow triggers

aniketdivekar committed:
fixed docker image tag

chrypnotoad committed:
add feature branch to test build

chrypnotoad committed:
Revert "add feature branch to test build"

This reverts commit cbb5082458e9ee2be6ce78a60ba33010228981c8.
